.github/workflows/java-ci.yml

```diff
@@ -0,0 +1,27 @@
+name: Java CI
+
+on:
+  push:
+    branches:
+      - ci-setup
+  pull_request:
+    branches:
+      - ci-setup
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Set up JDK 23
+        uses: actions/setup-java@v3
+        with:
+          java-version: '23'
+          distribution: 'temurin'
+          cache: 'maven'
+
+      - name: Build with Maven
+        run: mvn -B verify
```

Comment:
why are you using 2 different files for CI? is this necessary? maybe it would be easier to create 1 file where all the checks will be performed? it is more readable and compact